File: solutions/2503-longest-subarray-with-maximum-bitwise-and/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
    let max = 0;
    let maxRun = 0;
    let currentRun = 0;

    for(const num of nums){
        if(num > max){
            max = num;
            maxRun = 1;
            currentRun = 1;
        }else if(num === max){
            currentRun += 1;
            if(currentRun > maxRun){
                maxRun = currentRun
            }
        }else{
            currentRun = 0
        }

    }
    
    return maxRun;
};

File: solutions/2503-longest-subarray-with-maximum-bitwise-and/solution.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        maximum = max(nums)

        i = 0
        current = 0
        best = 0
        while i < len(nums):
            if nums[i] == maximum:
                current += 1
                best = max(best, current)
            else:
                current = 0
            i += 1

        return best
